LPS-80976 Introduce low level API to directly interact with the search engine

arboliveira opened the pull request:
Resending https://github.com/brianchandotcom/liferay-portal/pull/58800, addressing concerns from https://github.com/brianchandotcom/liferay-portal/pull/58800#issuecomment-391144572. Rewrote `ClusterRequestExecutorFixture`, and all other instantiation-intensive test fixtures, to comply with prior convention (particularly, "assign protected fields with an anonymous instance initializer block, not directly to variable.") Also did read all of the code and adjusted package and variable names, and other minor conventions.

Cleared `sf` and `relevant`; full CI shouldn't be needed. All new code for a new API, not used anywhere yet.

Author: @dylan-rebelak
Reviewers: @mhan810 @arboliveira

https://issues.liferay.com/browse/LPS-80976

Commit:
LPS-80976 portal-search-engine-adapter-api: add lower level API for interacting with search engine

https://issues.liferay.com/browse/LPS-80976

Co-authored-by: dylan-rebelak <dylan.rebelak@liferay.com>
Co-authored-by: Michael Han <michael.han@liferay.com>
Co-authored-by: André de Oliveira <andre.oliveira@liferay.com>

Commit:
LPS-80976 portal-search-elasticsearch6-impl: add cluster requests

https://issues.liferay.com/browse/LPS-80976

Co-authored-by: dylan-rebelak <dylan.rebelak@liferay.com>
Co-authored-by: Michael Han <michael.han@liferay.com>
Co-authored-by: André de Oliveira <andre.oliveira@liferay.com>

Commit:
LPS-80976 portal-search-elasticsearch6-impl: add document request impl

https://issues.liferay.com/browse/LPS-80976

Co-authored-by: dylan-rebelak <dylan.rebelak@liferay.com>
Co-authored-by: Michael Han <michael.han@liferay.com>
Co-authored-by: André de Oliveira <andre.oliveira@liferay.com>

Commit:
LPS-80976 portal-search-elasticsearch6-impl: add index requests impl

https://issues.liferay.com/browse/LPS-80976

Co-authored-by: dylan-rebelak <dylan.rebelak@liferay.com>
Co-authored-by: Michael Han <michael.han@liferay.com>
Co-authored-by: André de Oliveira <andre.oliveira@liferay.com>

Commit:
LPS-80976 portal-search-elasticsearch6-impl: add search request impl

https://issues.liferay.com/browse/LPS-80976

Co-authored-by: dylan-rebelak <dylan.rebelak@liferay.com>
Co-authored-by: Michael Han <michael.han@liferay.com>
Co-authored-by: André de Oliveira <andre.oliveira@liferay.com>

Commit:
LPS-80976 portal-search-elasticsearch6-impl: add new search engine adapter impl

https://issues.liferay.com/browse/LPS-80976

Co-authored-by: dylan-rebelak <dylan.rebelak@liferay.com>
Co-authored-by: Michael Han <michael.han@liferay.com>
Co-authored-by: André de Oliveira <andre.oliveira@liferay.com>

Commit:
LPS-80976 portal-search-elasticsearch6-impl: add tests

https://issues.liferay.com/browse/LPS-80976

Co-authored-by: dylan-rebelak <dylan.rebelak@liferay.com>
Co-authored-by: Michael Han <michael.han@liferay.com>
Co-authored-by: André de Oliveira <andre.oliveira@liferay.com>

arboliveira committed:
LPS-80976 portal-search-elasticsearch6-impl: build.gradle: We will implement portal-search-engine-adapter-api

https://issues.liferay.com/browse/LPS-80976

Co-authored-by: dylan-rebelak <dylan.rebelak@liferay.com>
Co-authored-by: Michael Han <michael.han@liferay.com>
Co-authored-by: André de Oliveira <andre.oliveira@liferay.com>

liferay-continuous-integration commented:
CI is automatically triggering &quot;ci:test:sf&quot; and &quot;ci:test:relevant&quot; for this pull to run Source Formatter and relevant tests.

Comment &quot;ci:test&quot; to run the full PR Tester for this pull.

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-20.liferay.com/job/test-portal-source-format/47/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-20.liferay.com//userContent/jobs/test-portal-source-format/builds/47/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest(master)/2496/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-17.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/2496/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 1 minute 51 seconds 699 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/230f42bd85188e5c3fffa8b25345f63e13946f4a">230f42bd85188e5c3fffa8b25345f63e13946f4a</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/arboliveira/liferay-portal/tree/LPS-80976-Low-level-Search-API">LPS-80976-Low-level-Search-API</a><br/>Branch GIT ID: <a href="https://github.com/arboliveira/liferay-portal/commit/931f81fa3b66a75110aed8ed72e6692b94fc76ae">931f81fa3b66a75110aed8ed72e6692b94fc76ae</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-20.liferay.com/job/test-portal-source-format/47/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-20.liferay.com/userContent/jobs/test-portal-source-format/builds/47/jenkins-report.html">here</a>.</h5></details></html>

arboliveira commented:
Resending https://github.com/brianchandotcom/liferay-portal/pull/58800, addressing concerns from https://github.com/brianchandotcom/liferay-portal/pull/58800#issuecomment-391144572. Rewrote `ClusterRequestExecutorFixture`, and all other instantiation-intensive test fixtures, to comply with prior convention (particularly, "assign protected fields with an anonymous instance initializer block, not directly to variable.") Also did read all of the code and adjusted package and variable names, and other minor conventions.

Full CI shouldn't be needed. All new code for a new API, not used anywhere yet.

Author: @dylan-rebelak
Reviewers: @mhan810 @arboliveira

https://issues.liferay.com/browse/LPS-80976

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:relevant - 9 out of 9 jobs passed in 57 minutes 27 seconds 197 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/230f42bd85188e5c3fffa8b25345f63e13946f4a">230f42bd85188e5c3fffa8b25345f63e13946f4a</a></p><h4>Copied in Private Modules Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal-ee/tree/master-private">master-private</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal-ee/commit/421cf6b41ca874678b98a9c05baecd0ae98a0b73">421cf6b41ca874678b98a9c05baecd0ae98a0b73</a></p>9 out of 9 jobs PASSED<details><summary><strong>9 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest(master)/2496/">test-portal-acceptance-pullrequest(master)</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/234491/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/237304/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/237305/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189141/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/86534/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/189142/">test-portal-acceptance-pullrequest-batch(master)/portal-frontend-js-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/84874/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-dist(master)/2501/">test-portal-acceptance-pullrequest-dist(master)/dist-jdk8</a></li></ul></details><h5>For more details click <a href="https://test-1-17.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/2496/jenkins-report.html">here</a>.</h5></details></html>

brianchandotcom commented:
Merged. Thank you.
View total diff: https://github.com/brianchandotcom/liferay-portal/compare/47af3cf387...f14a060dfa

brianchandotcom commented:
@arboliveira @mhan810 @dylan-rebelak please see my changes in upstream for this pull. Thx. Just some tiny changes..
